### PR TITLE
Reformed inventory operations, and fixed bugs

### PR DESCRIPTION
**Pull request by sakipvp**

### CODE REVIEW

File: Library/RSBot.Core/Objects/InventoryItem.cs [hunk 1]
```diff
@@ -343,5 +343,57 @@ public bool CanBeEquipped()
 
             return true;
         }
+
+        /// <summary>
+        /// Creates a clone from this.
+        /// </summary>
+        /// <returns>The clone <see cref="InventoryItem"/> instance.</returns>
+        public InventoryItem CreateClone()
```
Comment (ngoedde) [inline Library/RSBot.Core/Objects/InventoryItem.cs:351]: Did you consider implementing IClonable? Might be useful for other Classes, too. https://dotnetcoretutorials.com/2020/09/09/cloning-objects-in-c-and-net-core/ I like the idea of a CloningService which is called in the IClonable.Clone() method.

File: Library/RSBot.Core/Network/Handler/Agent/Inventory/InventoryOperationResponse.cs [hunk 2]
```diff
@@ -43,24 +43,24 @@ public void Invoke(Packet packet)
 
             switch (type)
             {
-                case 0:
-                    ParseInventoryToInventory(packet);
-                    //e.g when equipping a Bow (see ammo)
+                case 0: // InventoryToInventory
```
Comment (ngoedde) [inline Library/RSBot.Core/Network/Handler/Agent/Inventory/InventoryOperationResponse.cs:46]: Can you create an enum for the type?

File: Library/RSBot.Core/Objects/InventoryItem.cs (hunk 1 above)
Comment (sakipvp) [reply Library/RSBot.Core/Objects/InventoryItem.cs:351]: I thought about it. I prefer the ICloneable<T> interface with the T Clone () method.

**Comment by sakipvp** [comment]
I just see now, that the old methods were static. So make the new methods static too?

File: Library/RSBot.Core/Network/Handler/Agent/Inventory/InventoryOperationResponse.cs (hunk 2 above)
Comment (SDClowen) [reply Library/RSBot.Core/Network/Handler/Agent/Inventory/InventoryOperationResponse.cs:46]: I will do it.

**Comment by SDClowen** [comment]
> I just see now, that the old methods were static. So make the new methods static too?

I am working on inventory system.

**Comment by sakipvp** [comment]
You know, my master :-)
Good brainstorming!

**Pull request merged**